src/backend/aspen/app/views/sample.py

```diff
@@ -416,3 +416,66 @@ def create_sample():
     except Exception as e:
         raise ex.BadRequestException(f"Error encountered when saving data: {e}")
     return jsonify(success=True)
+
+
+@application.route("/api/samples/update/publicids", methods=["POST"])
+@requires_auth
+def update_sample_public_ids():
+    user: User = g.auth_user
+    request_data = request.get_json()
+
+    if not user.system_admin:
+        raise ex.BadRequestException(
+            "user making update request must be a system admin",
+        )
+
+    request_private_ids: list[str] = request_data["private_ids"]
+    request_public_ids: list[str] = request_data["public_ids"]
+    private_to_public: Mapping[str, str] = {
+        private: public
+        for (private, public) in zip(request_private_ids, request_public_ids)
+    }
+
+    group_id: int = request_data.get("group_id")
+
+    # group id not supplied (when admin is making request), use the user.group_id instead
```

Comment:
I might be totally wrong here -- I'm missing a lot of general knowledge and context, so grain of salt and all that -- but I'm worried about the group_id falling through to the user's group when it's not explicitly passed in the request.

Looking at the CLI command, the `--group-id` value is required. If we only expect this endpoint to be hit via the CLI, it seems to me like the endpoint should match and make it required.

On the other hand, if this should get hit outside the context of the CLI, does it make sense to default to the user's group_id? [That's a real question BTW, I just don't know enough] Only system admins should be able to call this end point, right? Does a system admin's group correspond with the samples we would be changing private/public IDs on? I would figure that the `submitting_group_id` would be a DPH group and that admins would not be in that group. And if I understand what comes next, that would then cause it to raise on `missing_private_identifiers`, but it wouldn't be that the IDs are missing, it's that the group_id fetch locked them out because it used the admin group unexpectedly.

Reply:
so initially i was thinking that i would future proof this by allowing a user to update their own public ids, and would therefore not need to provide a group id. I was thinking that the only time when the group_id would be passed would be when a system admin is updating a groups public_ids. I realize that this is confusing now that we're restricting this to only system admins.. I'll take defaulting to the group_id out, since this isn't an existing use case and would cause the unclear missing_private_identifiers error like what you were mentioning. That's a good call out.